internal/dex/dex.go

```diff
@@ -22,13 +23,33 @@ var (
 
 // OIDCProvider wraps an oidc.Provider and its Configuration
 type OIDCProvider struct {
+	Issuer   string
+	Endpoint *url.URL
 	Provider *oidc.Provider
 	Config   *oauth2.Config
 }
 
 // NewOIDCProvider construct an OIDCProvider fetching its configuration
 func NewOIDCProvider(ctx context.Context, issuer, clientID string) (*OIDCProvider, error) {
-	provider, err := oidc.NewProvider(ctx, issuer)
+	endpoint, err := url.Parse(issuer)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing the issuer URL")
+	}
+
+	return NewOIDCProviderWithEndpoint(ctx, issuer, clientID, endpoint)
+}
+
+// NewOIDCProviderWithEndpoint construct an OIDCProvider fetching its configuration from the endpoint URL
+func NewOIDCProviderWithEndpoint(ctx context.Context, issuer, clientID string, endpoint *url.URL) (*OIDCProvider, error) {
+	// If the issuer is different from the endpoint we need to set it in the context.
+	// With this differentiation the Epinio server can reach the Dex service through the Kubernetes DNS
+	// instead of the external URL. This was causing issues when the host was going to be resolved as a local IP (i.e: Rancher Desktop).
+	// - https://github.com/epinio/epinio/issues/1781
+	if issuer != endpoint.String() {
+		ctx = oidc.InsecureIssuerURLContext(ctx, issuer)
```

Comment:
[Reading this comment](https://pkg.go.dev/github.com/coreos/go-oidc/v3/oidc#InsecureIssuerURLContext):

```
This is insecure because validating the correct issuer is critical for multi-tenant proivders. Any overrides here MUST be carefully reviewed. 
```

I'm thinking, should we whitelist the allowed issuer? (e..g allow only the internal kube dns if they don't match)

Reply:
If we are not planning to support "external" Dex instances yes, we can check that the address has the svc.cluster.local domain in it. We can eventually drop it in the future.